opentasks-provider/src/main/java/org/dmfs/provider/tasks/processors/instances/Validating.java

```diff
@@ -43,50 +69,79 @@ public Validating(EntityProcessor<InstanceAdapter> delegate)
     public InstanceAdapter insert(SQLiteDatabase db, InstanceAdapter entityAdapter, boolean isSyncAdapter)
```

Comment:
Could this be refactored to something like this?
```java
    @Override
    public InstanceAdapter insert(SQLiteDatabase db, InstanceAdapter entityAdapter, boolean isSyncAdapter)
    {
        validateNotSyncadapter(isSyncAdapter);
        validateOnlyAllowedFieldsAreUpdated(entityAdapter);
        validateInstanceDoesNotExistAlready(db, entityAdapter);

        return mDelegate.insert(db, entityAdapter, false);
    }
```
The query is only serves as a validation as well, right?